protocols/data_processor.py

```
@@ -4,101 +4,105 @@
                             total_records += 1
                         processed_count += 1
-                        insights.append(f"{filename}: {type(data).__name__} with {len(data) if isinstance(data, (list, dict)) else 1} items")
-                except:
+                        insights.append(
+                            f"{filename}: {
+                                type(data).__name__} with {
+                                len(data) if isinstance(
+                                    data, (list, dict)) else 1} items"
+                        )
+                except BaseException:
                     pass
```

Ruff (0.11.9)

`try`-`except`-`pass` detected, consider logging the exception

(S110)